src/core/worker.lua

```diff
@@ -0,0 +1,132 @@
+-- worker.lua - Execute "worker" child processes to execute app networks
+-- Use of this source code is governed by the Apache 2.0 license; see COPYING.
+
+module(..., package.seeall)
+
+-- API:
+-- start(name, luacode)
+-- stop(name)
+-- status() -> table of { name = <info> }
+
+local lib = require("core.lib")
+local shm = require("core.shm")
+local S = require("syscall")
+
+--------------------------------------------------------------
+-- Master (parent) process code
+--------------------------------------------------------------
+
+local children = {}
+
+local function child (name)
+   return children[name] or error("no such child: " .. name)
+end
+
+-- Start a named worker to execute the given Lua code (a string).
+function start (name, luacode)
+   local pid = S.fork()
+   if pid == 0 then
+      -- First we perform some initialization functions and then we
+      -- restart the process with execv().
+      
+      -- Terminate automatically when the parent dies.
+      --
+      -- XXX This prctl setting needs to survive execve(). The Linux
+      -- execve(2) page seems to say that it will provided that the
+      -- binary being executed is not setuid or setgid. This may or
+      -- may not be adequate.
+      S.prctl("set_pdeathsig", "hup")
+      -- Symlink the shm "group" folder to be shared via the parent process.
+      shm.alias("group", "/"..S.getppid().."/group")
+      -- Save the code we want to run in the environment.
+      S.setenv("SNABB_PROGRAM_LUACODE", luacode, true)
+      -- Restart the process with execve().
+      -- /proc/$$/exe is a link to the same Snabb executable that we are running
+      local env = {}
+      for key, value in pairs(S.environ()) do
+         table.insert(env, key.."="..value)
+      end
+      S.execve(("/proc/%d/exe"):format(S.getpid()), {}, env)
+   else
+      -- Parent process
+      children[name] = { pid = pid }
```

Comment:
Is this the supervisor of the worker or the worker itself?  We need a way to get the worker PID.